### PR TITLE
SONAR-7543 Sanitize metric domains

### PR DESCRIPTION
**Pull request by stas-vilchik**

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/component-measures/home/reducer.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ const initialState = {
 };
 
 function groupByDomains (measures) {
-  const KNOWN_DOMAINS = ['Reliability', 'Security', 'Maintainability', 'Tests', 'Duplication', 'Size', 'Complexity'];
+  const KNOWN_DOMAINS = ['Reliability', 'Security', 'Maintainability', 'Coverage', 'Duplications', 'Size', 'Complexity'];
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/component-measures/home/reducer.js]: MINOR Split this 121 characters long line (which is greater than 120 authorized). [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3ALineLength)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 2]
```diff
@@ -53,6 +60,12 @@
   public static String DOMAIN_SCM = "SCM";
   public static String DOMAIN_ISSUES = "Issues";
   public static String DOMAIN_GENERAL = "General";
+  public static String DOMAIN_DUPLICATIONS = "Duplications";
+
+  /**
+   * @deprecated in 5.5. Renamed to {@link #DOMAIN_DUPLICATIONS}
+   */
+  @Deprecated
   public static String DOMAIN_DUPLICATION = "Duplication";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 3]
```diff
@@ -83,6 +96,11 @@
   public static String DOMAIN_TECHNICAL_DEBT = "Technical Debt";
 
   /**
+   * @since 5.5
+   */
+  public static String DOMAIN_RELEASABILITY = "Releasability";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_RELEASABILITY" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MAJOR Make DOMAIN_RELEASABILITY a static final constant or non-public and provide accessors if needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
MINOR Rename this field "DOMAIN_RELEASABILITY" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 4]
```diff
@@ -36,15 +36,22 @@
 
   // the following fields are not final to avoid compile-time constants used by plugins
   public static String DOMAIN_SIZE = "Size";
+  public static String DOMAIN_COVERAGE = "Coverage";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_COVERAGE" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MINOR Rename this field "DOMAIN_COVERAGE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 5]
```diff
@@ -53,6 +60,12 @@
   public static String DOMAIN_SCM = "SCM";
   public static String DOMAIN_ISSUES = "Issues";
   public static String DOMAIN_GENERAL = "General";
+  public static String DOMAIN_DUPLICATIONS = "Duplications";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: MAJOR Make this "public static DOMAIN_DUPLICATIONS" field final [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1444)
MAJOR Make DOMAIN_DUPLICATIONS a static final constant or non-public and provide accessors if needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)
MINOR Rename this field "DOMAIN_DUPLICATIONS" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS3008)

File: sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java [hunk 6]
```diff
@@ -36,15 +36,22 @@
 
   // the following fields are not final to avoid compile-time constants used by plugins
   public static String DOMAIN_SIZE = "Size";
+  public static String DOMAIN_COVERAGE = "Coverage";
+
+  /**
+   * @deprecated in 5.5. Merged into {@link #DOMAIN_COVERAGE}
+   */
+  @Deprecated
   public static String DOMAIN_TESTS = "Tests";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/measures/CoreMetrics.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 11 issues
* MAJOR 5 major
* MINOR 4 minor
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
LGTM

**Pull request merged**